blackbird_python/blackbird/program.py

```diff
@@ -206,9 +207,26 @@ def __call__(self, **kwargs):
         if not self.parameters:
             raise ValueError("Program is not a template!")
 
-        prog = copy.copy(self)
+        prog = copy.deepcopy(self)
```

Comment:
How come this was changed to a deepcopy?

Reply:
`copy` was the reason that the template BB program actually changed when calling it.
```
bs2 = bs1(**template_variables)
```
would change the elements in variable arrays in `bs1`, which it shouldn't. `deepcopy` fixes this.

Reply:
Nice, so a bugfix! Might be worth adding a separate entry to the changelog (is there a changelog in this repo?)

Reply:
There's no changelog; but I just added one. :slightly_smiling_face: